ref_builder/plan.py

```diff
@@ -28,34 +42,70 @@ def __str__(self) -> str:
         return f"{self.prefix} {self.key}"
 
 
-class Segment(BaseModel):
-    """The metadata of the segment."""
+class SequencePlan(BaseModel):
```

Comment:
This is describing segments, not sequences or plans, so I'd go with the name `SegmentMetadata`.